packages/gatsby-image/src/index.tsx

```diff
@@ -37,12 +170,21 @@ const convertProps = props => {
     convertedProps.loading = `eager`
   }
 
+  if (critical) {
+    logDeprecationNotice(`critical`, `the native "loading" attribute`)
+    convertedProps.loading = `eager`
+  }
+
   // convert fluid & fixed to arrays so we only have to work with arrays
   if (convertedProps.fluid) {
-    convertedProps.fluid = groupByMedia([].concat(convertedProps.fluid))
+    convertedProps.fluid = groupByMedia(
+      ([] as IFluidObject[]).concat(convertedProps.fluid)
+    ) as IFluidObject[]
```

Comment:
Thank you.
I had to cast `convertedProps.fluid` as `as`, since `convertedProps` is actually inferred to be the type of `IGatsbyImageProps`.
I fixed it by assigning it to another variable instead of `convertedProps.fluid` and returning the concatenated props.